rust/module.rs

```diff
@@ -170,17 +186,51 @@ fn build_modinfo_string_param(module: &str, field: &str, param: &str, content: &
 ///     author: b"Rust for Linux Contributors",
 ///     description: b"My very own kernel module!",
 ///     license: b"GPL v2",
-///     params: {},
+///     params: {
+///        my_i32: i32 {
+///            default: 42,
+///            permissions: 0o000,
+///            description: b"Example of i32",
+///        },
+///        writeable_i32: i32 {
+///            default: 42,
+///            permissions: 0o644,
+///            description: b"Example of i32",
+///        },
+///    },
 /// }
 ///
 /// struct MyKernelModule;
 ///
 /// impl KernelModule for MyKernelModule {
 ///     fn init() -> KernelResult<Self> {
+///         // If the parameter is writeable, then the kparam lock must be
+///         // taken to read the parameter:
+///         {
+///             let lock = THIS_MODULE.kernel_param_lock();
+///             println!("i32 param is:  {}", writeable_i32.read(&lock));
+///         }
+///         // If the parameter is read only, it can be read without locking
+///         // the kernel parameters:
+///         println!("i32 param is:  {}", my_i32.read());
 ///         Ok(MyKernelModule)
 ///     }
 /// }
 /// ```
+///
+/// ## Suported Parameter Types
+///
+/// - `bool`    - Corresponds to C `bool` param type.
+/// - `u8`      - Corresponds to C `char` param type.
+/// - `i16`     - Corresponds to C `short` param type.
+/// - `u16`     - Corresponds to C `ushort` param type.
+/// - `i32`     - Corresponds to C `int` param type.
+/// - `u32`     - Corresponds to C `uint` param type.
+/// - `u64`     - Corresponds to C `ullong` param type.
+/// - `invbool` - Corresponds to C `invbool` param type
+///               Value is inverse of user input.
```

Comment:
Why would anyone want this?

Reply:
From a quick look, it seems it has added when the module param functionality was rewritten back in 2002. More importantly, only 2 fbdev modules use it (and from the log, it seems it only ever had 2 or so users too). And one of them is for the ATI Rage128... :-)

Thus it looks like a "feature" that never got used. So, yeah, it is best to not introduce it in the Rust side too; and should probably be proposed to be removed from the C side too.

Reply:
Funny - I saw it in "Linux Device Drivers" and just assumed it was in use. Removed it now.